crates/astria-sequencer/src/mempool.rs

```diff
@@ -62,13 +67,18 @@ impl PartialOrd for TransactionPriority {
 pub(crate) struct EnqueuedTransaction {
     tx_hash: [u8; 32],
     signed_tx: Arc<SignedTransaction>,
+    ttl: i64,
 }
 
 impl EnqueuedTransaction {
-    fn new(signed_tx: SignedTransaction) -> Self {
+    fn new(signed_tx: SignedTransaction, ttl: i64) -> Self {
         Self {
             tx_hash: signed_tx.sha256_of_proto_encoding(),
             signed_tx: Arc::new(signed_tx),
+            ttl: Time::now()
+                .unix_timestamp()
+                .checked_add(ttl)
+                .expect("overflow in enqueued transaction ttl"),
```

Comment:
i'd prefer having a `timeout_time` passed in, which is a unix timestamp of when the tx expires, over a ttl. it'll be easier to test and compute client-side when the exact timeout is going to be before submitting

Reply:
would the time be decided by the transaction signer? I'm a little hesitant to let users add an expiration if they aren't signing over the expiration. future block builders wouldn't have to respect non-signed data if they didn't want to

Reply:
okay right, since this doesn't actually add it to the tx type, this is fine